prov/shm/src/smr_ep.c

```diff
@@ -302,17 +307,102 @@ void smr_format_iov(struct smr_cmd *cmd, fi_addr_t peer_id,
 		    const struct iovec *iov, size_t count, size_t total_len,
 		    uint32_t op, uint64_t tag, uint64_t data, uint64_t op_flags,
 		    void *context, struct smr_region *smr,
-		    struct smr_resp *resp, struct smr_cmd *pend_cmd)
+		    struct smr_resp *resp, struct smr_tx_entry *pend)
 {
 	smr_generic_format(cmd, peer_id, op, tag, 0, 0, data, op_flags);
 	cmd->msg.hdr.op_src = smr_src_iov;
 	cmd->msg.hdr.src_data = (uintptr_t) ((char **) resp - (char **) smr);
 	cmd->msg.data.iov_count = count;
 	cmd->msg.hdr.size = total_len;
-	cmd->msg.hdr.msg_id = (uint64_t) (uintptr_t) context;
 	memcpy(cmd->msg.data.iov, iov, sizeof(*iov) * count);
 
-	smr_post_pend_resp(cmd, pend_cmd, resp);
+	smr_post_pend_resp(pend, cmd, context, iov, count, resp);
+}
+
+void smr_format_mmap(struct smr_cmd *cmd, fi_addr_t peer_id,
+		     const struct iovec *iov, size_t count, size_t total_len,
+		     uint32_t op, uint64_t tag, uint64_t data, uint64_t op_flags,
+		     void *context, struct smr_region *smr, uint64_t msg_id,
+		     struct smr_ep_name *map_name, void *shm_ptr,
+		     struct smr_resp *resp, struct smr_tx_entry *pend)
+{
+	smr_generic_format(cmd, peer_id, op, tag, 0, 0, data, op_flags);
+	cmd->msg.hdr.op_src = smr_src_mmap;
+	cmd->msg.hdr.msg_id = msg_id;
+	cmd->msg.hdr.src_data = (uintptr_t) ((char **) resp - (char **) smr);
+	cmd->msg.hdr.size = total_len;
+
+	smr_post_pend_resp(pend, cmd, context, iov, count, resp);
+
+	pend->map_name = map_name;
+	pend->map_ptr = shm_ptr;
+}
+
+int smr_iov_mmap_copy_in(struct smr_ep *ep, struct smr_region *peer_smr,
+			 const struct iovec *iov, size_t count,
+			 size_t total_len, uint32_t op, uint64_t msg_id,
+			 struct smr_ep_name **map_name, void **shm_ptr)
+{
+	char peer_name[NAME_MAX];
+	char shm_name[NAME_MAX];
+	void *mapped_ptr;
+	int fd, ret, num;
+
+	sprintf(peer_name, "%s", (char *)peer_smr + peer_smr->name_offset);
+	num = sprintf(shm_name, "%s_%s_%ld", ep->name, peer_name, msg_id);
```

Comment:
Please use snprintf, so we don't immediately need to respond to static code analysis complaining about possible overflow.

Reply:
Sure. Updated with snprintf.